internal/expr/typeinfo.go

```diff
@@ -9,6 +9,9 @@ import (
 	"sync"
 )
 
+// typeMember should be implemented without pointer receivers as it is used as
+// a key in maps and, even if the underlying structs are equal, the pointers
+// may not be.
```

Comment:
I'm probably missing something here. These values were designed to be a long term cache of type data, and as such using them as pointers should be fine. In the original design, it also meant that these values were always unique, so we might even use them as keys in a map, and that's a pretty good key since this is a pointer.

Now it sounds like this has changed, maybe after introducing `mapKey`?  But this outcome sounds a bit like previous conversations we had: it's an unintended design change that is spreading abstraction breaking issues.  I would instead prefer to restore the original intention of this logic: to cache struct information.

Just as some backing evidence for this, the data that composes the `structField` type below looks like a pretty bad way to have a unique map key.

Happy to discuss this further.